src/librustc_typeck/collect.rs

```diff
@@ -1359,7 +1359,9 @@ fn type_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId) -> Ty<'tcx> {
                                         param_index += 1;
                                     }
                                 }
-                                bug!("there was no generic parameter associated with the arg");
+                                // This is no generic parameter associated with the arg. This is probably from
+                                // an extra arg where one is not needed.
+                                return tcx.types.err;
```

Comment:
Did the `bug` actually get triggered with invalid args?

Reply:
Yes, it was getting triggered even after printing the errors created in the previous commit.